### PR TITLE
SimulationRenderer: Fix COLOR_ATTACHMENT format bug in Firefox

### PR DESCRIPTION
**Pull request by Mugen87**
The example [webgl_gpgpu_birds](http://threejs.org/examples/#webgl_gpgpu_birds) is not working in Firefox. The following error message is logged in the browser console:

_Error: WebGL: drawElements: Incomplete framebuffer: Status 0x8cd6: COLOR_ATTACHMENT0 has an effective format of RGB32F, which is not renderable._

This PR changes the format of the velocity render target from `THREE.RGBFormat` to `THREE.RGBAFormat` to fix the bug. 

Tested with FF 45.0.1, Chrome 49.0.2623.87 and Safari 9.0.3 on OS X 10.10.5

### CODE REVIEW

File: examples/js/SimulationRenderer.js [hunk 1]
```diff
@@ -102,7 +102,7 @@ function SimulationRenderer( WIDTH, renderer ) {
 
 		rtPosition1 = getRenderTarget( THREE.RGBAFormat );
 		rtPosition2 = rtPosition1.clone();
-		rtVelocity1 = getRenderTarget( THREE.RGBFormat );
+		rtVelocity1 = getRenderTarget( THREE.RGBAFormat );
```
Comment (WestLangley) [inline examples/js/SimulationRenderer.js]: Is this change consistent with `THREE.RGBFormat` in `generateVelocityTexture()`?

**Comment by Mugen87** [comment]
I think the format of the `DataTexture` should be `THREE.RGBFormat`. If i change the format to `THREE.RGBAFormat`,  Firefox says: _Error: WebGL: texImage2D: Provided buffer is too small. (needs 16384, has 12288)_

To avoid this message one must enlarge the typed array in `generateVelocityTexture`, what is actually not necessary.  

```javascript
//var a = new Float32Array( PARTICLES * 3 );
var a = new Float32Array( PARTICLES * 4 ); // useless
```

**Comment by WestLangley** [comment]
Why is FF complaining? Is the complaint a valid one?

If it is, then the code needs to be modified so the initial velocity data is populated correctly.

/ping @zz85

**Comment by Mugen87** [comment]
> Why is FF complaining?

In FF source code, there is a method called [TexUnpackBytes::ValidateUnpack](https://hg.mozilla.org/releases/mozilla-aurora/file/tip/dom/canvas/TexUnpackBlob.cpp#l52) which is used in [WebGLTexture::TexImage](https://hg.mozilla.org/releases/mozilla-aurora/file/tip/dom/canvas/WebGLTextureUpload.cpp#l1094) to validate textures. 

There a two checks in this method. One check logs the mentioned error message.

```c++
if (mByteCount < bytesNeeded.value()) {

        webgl->ErrorInvalidOperation("%s: Provided buffer is too small. (needs %u, has"
                                     " %u)",
                                     funcName, bytesNeeded.value(), mByteCount);
        return false;
}
```

**Comment by Mugen87** [comment]
@WestLangley The last commit should correct the initial data provision. Do you think it's okay like this?

**Comment by WestLangley** [comment]
@Mugen87 Yes. That is what I was expecting you to do.

You should be able to verify the initialization is correct by hardwiring the birds to, say 5, making sure you can see all of them, and comment out the updating of the velocities. All the birds should fly in a straight line, maintaining their initial heading and speed.

I still would like to know which line of code causes the problem when the velocity buffer is not RGBA...

**Comment by Mugen87** [comment]
> All the birds should fly in a straight line, maintaining their initial heading and speed.

They actually do this :wink:

**Comment by Mugen87** [comment]
> I still would like to know which line of code causes the problem when the velocity buffer is not RGBA...

The initial error messages were logged after these calls [here](https://github.com/mrdoob/three.js/blob/master/src/renderers/webgl/WebGLIndexedBufferRenderer.js#L35) and [here](https://github.com/mrdoob/three.js/blob/master/src/renderers/WebGLRenderer.js#L470) but only if the render target was the velocity-render-target.

**Comment by WestLangley** [comment]
@mrdoob This can be merged.

@Mugen87 @zz85 If you can figure out why the velocity buffer must be RGBA -- or if it really must be -- I would be grateful.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**